cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc

```diff
@@ -245,51 +697,54 @@ An example of this is `CALL db.index.explicit.searchNodes('my_index','email:me*'
+| Feature     | Type | Change | Details
+| `rels()`    | Function  | Deprecated | Replaced by <<functions-relationships,relationships()>>
+| `toInt()`   | Function  | Deprecated | Replaced by <<functions-tointeger,toInteger()>>
+| `lower()`   | Function  | Deprecated | Replaced by <<functions-tolower,toLower()>>
+| `upper()`   | Function  | Deprecated | Replaced by <<functions-toupper,toUpper()>>
 | <<functions-toboolean,toBoolean()>> | Function | Added |
-| <<cypher-map-projection, Map projection>> | Syntax | Added |
-| <<cypher-pattern-comprehension, Pattern comprehension>> | Syntax | Added |
-| <<java-reference#extending-neo4j-functions, User-defined functions>> | Functionality | Added |
-| <<query-call, CALL\...YIELD\...WHERE>>   | Clause  | Extended  | Records returned by `YIELD` may be filtered further using `WHERE`
+| <<cypher-map-projection,Map projection>> | Syntax | Added |
+| <<cypher-pattern-comprehension,Pattern comprehension>> | Syntax | Added |
+| <<java-reference#extending-neo4j-functions,User-defined functions>> | Functionality | Added |
+| <<query-call,CALL\...YIELD\...WHERE>>   | Clause  | Extended  | Records returned by `YIELD` may be filtered further using `WHERE`
 |===
 
 
 [[cypher-deprecations-additions-removals-3.0]]
-=== Version 3.0
+== Version 3.0
 [options="header"]
 |===
 | Feature          | Type | Change | Details
-| `has()`  | Function  | Removed | Replaced by <<functions-exists, exists()>>
-| `str()`  | Function  | Removed | Replaced by <<functions-tostring, toString()>>
-| `+{parameter}+` | Syntax | Deprecated | Replaced by <<cypher-parameters, $parameter>>
-| <<functions-properties, properties()>>  | Function  | Added  |
+| `has()`  | Function  | Removed | Replaced by <<functions-exists,exists()>>
+| `str()`  | Function  | Removed | Replaced by <<functions-tostring,toString()>>
+| `+{parameter}+` | Syntax | Deprecated | Replaced by <<cypher-parameters,$parameter>>
+| <<functions-properties,properties()>>  | Function  | Added  |
 | <<query-call,CALL [\...YIELD]>>   | Clause  | Added  |
 | <<functions-point-cartesian-2d,point() - Cartesian 2D>> | Function | Added |
 | <<functions-point-wgs84-2d,point() - WGS 84 2D>> | Function | Added |
 | <<functions-distance,distance()>> | Function | Added |
-| <<java-reference#extending-neo4j-procedures, User-defined procedures>> | Functionality | Added |
-| <<functions-tostring, toString()>>   | Function  | Extended | Now also allows Boolean values as input
+| <<java-reference#extending-neo4j-procedures,User-defined procedures>> | Functionality | Added |
+| <<functions-tostring,toString()>>   | Function  | Extended | Now also allows Boolean values as input
 |===
 
 
 [[cypher-compatibility]]
 == Compatibility
 
-NOTE: The ability of Neo4j to support multiple older versions of the Cypher language has been changing.
+[NOTE]
+====
+The ability of Neo4j to support multiple older versions of the Cypher language has been changing.
```

Comment:
This note is really awkward. I tried to rewrite it, but please make sure I haven't changed the meaning:
_Neo4j’s ability to support multiple older versions of the Cypher language has changed over time. 
In versions prior to Neo4j 3.4, the backwards compatibility layer included the Cypher language parser, planner, and runtime. All supported versions of Cypher ran on the same Neo4j kernel. However, this changed in Neo4j 3.4 when the runtime was excluded from the compatibility layer. When you run, e.g. a CYPHER 3.1 query in Neo4j 3.5, the query is planned with the 3.1 planner, but run with 3.5 runtime and kernel. 
The compatibility layer changed again in Neo4j 4.0 and it now includes only the parser. When you run a CYPHER 3.5 query, e.g., in 4.3, Neo4j parses the older language features, but uses the 4.3 planner, runtime, and kernel to plan and run the query.
The primary reason for these changes is the optimizations in the Cypher runtime to allow Cypher queries to perform better._

Reply:
Wow great rewrite thanks.